src/Provider/TaxRateProvider.php

```diff
@@ -5,26 +5,23 @@
 namespace Sylius\RefundPlugin\Provider;
 
 use Doctrine\Common\Collections\Collection;
-use Sylius\Component\Core\Model\AdjustmentInterface;
 use Sylius\Component\Core\Model\OrderItemUnitInterface;
+use Sylius\RefundPlugin\Entity\AdjustmentInterface;
+use Webmozart\Assert\Assert;
 
-final class LabelBasedTaxRateProvider implements TaxRateProviderInterface
+final class TaxRateProvider implements TaxRateProviderInterface
 {
     public function provide(OrderItemUnitInterface $orderItemUnit): ?string
     {
         /** @var Collection|AdjustmentInterface[] $taxAdjustments */
         $taxAdjustments = $orderItemUnit->getAdjustments(AdjustmentInterface::TAX_ADJUSTMENT);
 
-        if ($taxAdjustments->isEmpty()) {
-            return null;
-        }
+        Assert::maxCount($taxAdjustments, 1, 'Every Order Item Unit Should have max 1 tax adjustment');
```

Comment:
```suggestion
        Assert::maxCount($taxAdjustments, 1, 'Each Order Item Unit should not have more than 1 tax adjustment');
```
?
And I wonder if we shouldn't throw here a custom exception instead of using assertion, sth like `MoreThanOneTaxAdjustment`

Reply:
I thought that Assert will be enough but custom exception looks better

Reply:
Custom adjustment is a great idea. About the exception message in it - what would you say about: "'Each Order Item Unit must not have more than 1 tax adjustment'"?